### PR TITLE
NVDA source code and user guide: remove Windows 8 constant mentions

### LINKED ISSUE #15662: Ease of Access: deprecate canConfigTerminateOnDesktopSwitch as it is always True
Hi,

Part of #15167 and #15647:

### Is your feature request related to a problem? Please describe.
As of NVDA 2024.1 alpha, Python 3.11 is in use, meaning Windows 8.1 or later is required. At the moment easeOfAccess.canConfigTerminateOnDesktopSwitch requires Windows 8, and with NVDA 2024.1, that flag is alwasy set to True.

### Describe the solution you'd like
Deprecate easeOfAccess.canConfigTerminateOnDesktopSwitch, always returing "True". This allows code relying on this flag - namely config module - to be edited based on deprecation warning, along with giving add-ons community time to adjust their code.

### Describe alternatives you've considered
Either leave code as is or remove the flag altogether.

### Additional context
easeOfAccess.canConfigTerminateOnDesktopSwitch uses winVersion.WIN8 constant (to be deprecated once #15647 comes to life).

Thanks.

### LINKED ISSUE #15663: Windows API: remove Windows 7 specific lock session structure/enumeration
Hi,

Stemming from #15167 and #15647:

### Is your feature request related to a problem? Please describe.
Due to value issues, separate Widnows session lock state enumerations are provided for Windows 7 and 8.x and later. Now that Windows 8.1 is the minimum OS, remove Windows 7 enumerations.

### Describe the solution you'd like
Remove Windows 7 lock state enumerations.

### Describe alternatives you've considered
Leave code as is.

### Additional context
ONe of the places where winVersion.WIN8 constant is used in NVDA code.

Thanks.

### LINKED ISSUE #15664: Screen curtain: assume full screen magnification is available (for now)
Hi,
Stemming from #15647:

### Is your feature request related to a problem? Please describe.
At the moment screen curtain relies on full screen magnification flag, which returns True if running on Windows 8 and later. Since NVDA requires Windows 8.1 now, there is no need for this flag but add-ons may use it for some time.

### Describe the solution you'd like
Deprecate winVersion.isFullScreenMagnificationAvailable flag, and in screen curtain module, return "True" in check method.

### Describe alternatives you've considered
Leave code as is.

### Additional context

Thanks.

### PR DESCRIPTION
**Pull request by josephsl**
### Link to issue number:
Closes #15663 
Closes #15664 

### Summary of the issue:
Remove mention of Windows 8 (winVersion.WIN8) constant from source code, and for the user guide, remove Windows 8 mention from audio ducking setting.

### Description of user facing changes
None

### Description of development approach
* winVersion: isFullScreenMagnificationAvailable will return Windows 8.1 or later.
* WTS lock session: removed private Windows 7 enumeration.
* User guide: edited audio ducking setting paragraph.

### Testing strategy:
Manual testing with code coverage - making sure NVDA runs on Windows 8.1 and later with screen curtain, Ease of Access, and session lock functionality tested.

### Known issues with pull request:
None

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### Deprecation notice:
easeOfAccess.canConfigTerminateOnDesktopSwitch is deprecated with no replacement - this wil always return True as NVDA now requires Windows 8.1 or later.

Thanks.

### CODE REVIEW

**Comment by josephsl** [comment]
Note: #15662 is closed thanks to #15644.

**Comment by seanbudd** [comment]
Should you remove "Closes #15662" from the PR description?

File: source/winVersion.py [hunk 1]
```diff
@@ -217,4 +217,4 @@ def isFullScreenMagnificationAvailable() -> bool:
 	WOW64 applications such as NVDA. For our usages, support has been added since Windows 8, relying on our
 	testing our specific usage of the API with each Windows version since Windows 8
 	"""
-	return getWinVer() >= WIN8
+	return getWinVer() >= WIN81
```
Comment (seanbudd) [inline source/winVersion.py:220]: Shouldn't this be deprecated or at the very least `return True`?

```suggestion
	return True
```

File: source/easeOfAccess.py [hunk 2]
```diff
@@ -166,5 +166,7 @@ def setAutoStart(autoStartContext: AutoStartContext, enable: bool) -> None:
 			0,
 			winreg.KEY_READ | winreg.KEY_WRITE | winreg.KEY_WOW64_64KEY
 		)
-		winreg.SetValueEx(k, "Configuration", None, winreg.REG_SZ,
-			",".join(conf))
+		winreg.SetValueEx(
+			k, "Configuration", None, winreg.REG_SZ,
+			",".join(conf)
+		)
```
Comment (seanbudd) [inline source/easeOfAccess.py:172]:
```suggestion
		winreg.SetValueEx(
			k,
			"Configuration",
			None,
			winreg.REG_SZ,
			",".join(conf)
		)
```

**Comment by josephsl** [comment]
Initial comment edited, thanks for the reminder.

File: source/winVersion.py [hunk 3]
```diff
@@ -217,4 +217,4 @@ def isFullScreenMagnificationAvailable() -> bool:
 	WOW64 applications such as NVDA. For our usages, support has been added since Windows 8, relying on our
 	testing our specific usage of the API with each Windows version since Windows 8
 	"""
-	return getWinVer() >= WIN8
+	return True
```
Comment (seanbudd) [inline source/winVersion.py:220]: i think it makes sense to deprecate this function to match the strategy used in #15644.
That way usage in NVDA core is discouraged and removed in favour of simpler code.

Comment (seanbudd) [reply source/winVersion.py:220]: i.e `ScreenCurtainProvider.canStart` can also be simplified to `return True`

File: user_docs/en/changes.t2t [hunk 4]
```diff
@@ -117,6 +117,7 @@ Code which imports from one of them, should instead import from the replacement
 
 - Using ``watchdog.getFormattedStacksForAllThreads`` is deprecated - please use ``logHandler.getFormattedStacksForAllThreads`` instead. (#15616, @lukaszgo1)
 - ``easeOfAccess.canConfigTerminateOnDesktopSwitch`` has been deprecated, as it became obsolete since Windows 7 is no longer supported. (#15644, @LeonarddeR)
+- ``winVersion.isFullScreenMagnificationAvailable`` has been deprecated, as it became obsolete since Windows 7 is no longer supported. (#15664, @josephsl)
```
Comment (seanbudd) [inline user_docs/en/changes.t2t:120]: this deprecation could also suggest `ScreenCurtainProvider.canStart` as an alternative.

**Comment by josephsl** [comment]
Hi, done. Thanks.

File: source/winVersion.py [hunk 5]
```diff
@@ -211,10 +213,14 @@ def isUwpOcrAvailable():
...
 
 
-def isFullScreenMagnificationAvailable() -> bool:
-	"""
-	Technically this is always False. The Magnification API has been marked by MS as unsupported for
-	WOW64 applications such as NVDA. For our usages, support has been added since Windows 8, relying on our
-	testing our specific usage of the API with each Windows version since Windows 8
-	"""
-	return getWinVer() >= WIN8
+if NVDAState._allowDeprecatedAPI():
```
Comment (seanbudd) [inline source/winVersion.py:216]: can this please use the standard `__getattr__` methoud outlined in [deprecations.md](https://github.com/nvaccess/nvda/blob/master/projectDocs/dev/deprecations.md#deprecations-1)?

File: source/winVersion.py [hunk 6]
```diff
@@ -211,10 +213,14 @@ def isUwpOcrAvailable():
...
+	def isFullScreenMagnificationAvailable() -> bool:
+		"""
+		Technically this is always False. The Magnification API has been marked by MS as unsupported for
+		WOW64 applications such as NVDA. For our usages, support has been added since Windows 8, relying on our
+		testing our specific usage of the API with each Windows version since Windows 8
+		"""
+		log.debugWarning(
+			"Deprecated function called: winVersion.isFullScreenMagnificationAvailable", stack_info=True
```
Comment (seanbudd) [inline source/winVersion.py:224]: this should suggest the alternative as well

File: source/winVersion.py (hunk 5 above)
Comment (josephsl) [reply source/winVersion.py:216]: I did it like this initially as the deprecated symbol is a function, not a constant or a variable, using gui.quit as precedence. If an add-on or other part of NVDA calls winVersion.isFullScreenMagnificationAvailable using the proposed deprecation method, Python will raise attribute error, which amounts to removing the function altogether from the module.

Comment (seanbudd) [reply source/winVersion.py:216]: Makes sense

Comment (seanbudd) [reply source/winVersion.py:216]: I think the `gui.quit` example did come before this was standardised though, so I wouldn't use it as precedent

Comment (josephsl) [reply source/winVersion.py:216]: I think this should give us a motivation to refine the getattr method when dealing with deprecated functions and class methods. I'll follow the suggested process, and then will create a new GitHub issue.

**Review by seanbudd — APPROVED** [review]
Thanks @josephsl

**Pull request merged**